pkg/controller/operators/operator_controller_test.go

```diff
@@ -204,6 +204,44 @@ var _ = Describe("Operator Controller", func() {
 				})
 			})
 
+			Context("when multiple types of a gvk are labeled", func() {
+				expectedServiceAccountOrder := []string{"sa-1", "sa-2", "sa-3", "sa-4", "sa-5", "sa-6", "sa-7", "sa-8", "sa-9"}
+				BeforeEach(func() {
+					newObjs := make([]runtime.Object, 10)
+					for i, saName := range expectedServiceAccountOrder {
+						// Create objects in reverse order to ensure they are eventually ordered
+						newObjs[9-i] = testobj.WithNamespacedName(
+							&types.NamespacedName{Namespace: namespace, Name: saName}, &corev1.ServiceAccount{},
+						)
+					}
+
+					for _, obj := range newObjs {
+						Expect(k8sClient.Create(ctx, obj.(client.Object))).To(Succeed())
+					}
+
+					objs = append(objs, newObjs...)
+					expectedRefs = append(expectedRefs, toRefs(scheme, newObjs...)...)
+				})
+
+				It("should list each of the component references in alphabetical order by namespace and name", func() {
+					Eventually(func() ([]operatorsv1.RichReference, error) {
+						err := k8sClient.Get(ctx, name, operator)
+						return operator.Status.Components.Refs, err
+					}, timeout, interval).Should(ConsistOf(expectedRefs))
+
+					for _, ref := range operator.Status.Components.Refs {
+						if ref.Kind != rbacv1.ServiceAccountKind {
+							continue
+						}
+
+						Expect(expectedServiceAccountOrder).Should(Not(BeEmpty()))
+						Expect(ref.Name).Should(Equal(expectedServiceAccountOrder[0]))
+						expectedServiceAccountOrder = expectedServiceAccountOrder[1:]
+					}
+					Expect(expectedServiceAccountOrder).Should(BeEmpty())
```

Comment:
Do we need an auxiliary type here? We have access to the loop iteration value, so we could capture that vs. using a blank identifier for the index value.

```go
for i, ref := range operator.Status.Components.Refs {
    if ref.Kind != rbacv1.ServiceAccountKind { continue }
    Expect(ref.Name).Should(Equal(fmt.Sprintf("sa-%d", i))
```

Thoughts?

Reply:
That seems reasonable, I still want to make sure that all the serviceAccounts are found, so I might make a slight modification:

```
serviceAccountCount := 0
					for i, ref := range operator.Status.Components.Refs {
						if ref.Kind != rbacv1.ServiceAccountKind {
							continue
						}

						Expect(ref.Name).Should(Equal(fmt.Sprintf("sa-%d", i)))
						serviceAccountCount++
					}
					Expect(serviceAccountCount).Should(Equal(9))
				})
```

Reply:
This doesn't work if there is any other object associated with the operator, had to update the code slightly:
```
serviceAccountCount := 0
for _, ref := range operator.Status.Components.Refs {
	if ref.Kind != rbacv1.ServiceAccountKind {
		continue
	}

	Expect(ref.Name).Should(Equal(fmt.Sprintf("sa-%d", serviceAccountCount)))
	serviceAccountCount++
}
Expect(serviceAccountCount).Should(Equal(9))
```

Reply:
Yep, that works for me.